includes/HttpFunctions.php

```diff
@@ -506,16 +506,11 @@ public function execute() {
 			$this->setUserAgent( Http::userAgent() );
 		}
 
-		// @author macbre
-		// pass Request ID to internal requests
-		$this->setHeader( Wikia\Util\RequestId::REQUEST_HEADER_NAME, Wikia\Util\RequestId::instance()->getRequestId() );
-		// PLATFORM-1473: pass X-Wikia-Internal-Request
-		$this->setHeader( WebRequest::WIKIA_INTERNAL_REQUEST_HEADER, 'mediawiki' );
-
 		// Wikia change - begin - @author: wladek
-		// Append extra headers for internal requests, currently only X-Request-Origin-Host
-		if ( $this->internalRequest ) {
-			$this->setHeader( Wikia\Util\RequestId::REQUEST_HEADER_ORIGIN_HOST, wfHostname() );
+		$tracerInstance = \Wikia\Tracer\WikiaTracer::instance();
```

Comment:
This API seems a bit complex, forcing us to use the public/internal logic outside the class.  Since setting headers is one of the things WikiaTracer is for, why not a helper function to do this.  Since setHeader just uses $reqHeader which is an array, we could pass that in as a variable.